lib/NMISNG/Node.pm

```diff
@@ -1609,30 +1618,32 @@ sub save
 						when => $saveTime) if ($audit_enabled);
 
 		
-		# This code creates/updates catchall on save, this has been commented out for the moment because
-		# adding a new node on a primary should not create a catchall on the primary (which this would do)
-
+		# This code creates/updates catchall on save, when the node cluster_id matches the server cluster_id
+		# (so it is being polled here)
 		# if we can lock the node then update host info and it's catchall, for new nodes this should work
 		# every time (which is what we are most concerned about)
 		# NOTE: this will create a catchall for a new node
-		# $self->nmisng->log->debug2(sub {"Getting lock for node to update host_addr and sync catchall $entry{name}"});
-		# my $lock = $self->lock(type => 'update');
-		# if( !$lock->{error} && !$lock->{conflict} ) 
-		# {
-		# 	my $path = $self->inventory_path(concept => "catchall", data => {}, path_keys => []);
-		# 	my ($catchall_inventory, $error) =  $self->inventory( concept => "catchall", path => $path, path_keys => [], create => 1 );
-		# 	if( !$error ) 
-		# 	{
-		# 		my $catchall_data = $catchall_inventory->data_live();
-		# 		$self->update_host_addr( catchall_data => $catchall_data );
-		# 		$self->sync_catchall( cache => $catchall_inventory );
-		# 		$self->unlock(lock => $lock);
-		# 	}
-		# 	else 
-		# 	{
-		# 		$self->nmisng->log->warn(sub {"Node::save failed to get catchall so cannot sync: $error"});	
-		# 	}
-		# }
+		if( $self->cluster_id eq $self->nmisng->config->{cluster_id} ) {
+			$self->nmisng->log->debug2(sub {"Getting lock for node to update host_addr and sync catchall $entry{name}"});
+			my $lock = $self->lock(type => 'update'); # we might want a collect lock here?
+			if( !$lock->{error} && !$lock->{conflict} ) 
+			{
+				my $path = $self->inventory_path(concept => "catchall", data => {}, path_keys => []);
+				my ($catchall_inventory, $error) =  $self->inventory( concept => "catchall", path => $path, path_keys => [], create => 1 );
+				if( !$error ) 
+				{
+					my $catchall_data = $catchall_inventory->data_live();
+					$self->update_host_addr( catchall_data => $catchall_data );
+					$self->sync_catchall( cache => $catchall_inventory );
+					$self->unlock(lock => $lock);
+				}
+				else 
+				{
+					# this isn't a huge deal
```

Comment:
just a small question, Should we keep the lock or should we unlock,  in case it fails $self->inventory() ??